packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -116,4 +126,86 @@ contract L1Block is ISemver {
             sstore(batcherHash.slot, calldataload(132)) // bytes32
         }
     }
+
+    /// @notice Updates the L1 block values for an Interop upgraded chain.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _baseFeeScalar      L1 base fee scalar
+    ///   2. _blobBaseFeeScalar  L1 blob base fee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _timestamp          L1 timestamp.
+    ///   5. _number             L1 blocknumber.
+    ///   6. _basefee            L1 base fee.
+    ///   7. _blobBaseFee        L1 blob base fee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    ///  10. _dependencySetSize  Size of the interop dependency set.
+    ///  11. _dependencySet      Array of chain IDs for the interop dependency set.
+    function setL1BlockValuesInterop() external {
+        bytes4 errorNotDepositorSelector = NotDepositor.selector;
+        bytes4 errorDependencySetSizeMismatch = DependencySetSizeMismatch.selector;
+        assembly {
+            // Revert if the caller is not the depositor account.
+            if xor(caller(), DEPOSITOR_ACCOUNT) {
+                mstore(0x00, errorNotDepositorSelector)
```

Comment:
I think we should just inline the selector here instead of defining it up above as `bytes4` because you cannot reference it in yul, defining the error type is more for having the error explicit in the abi so tooling can just work with it as well as being able to test that the error selector is correct